codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/error/ServiceErrorGenerator.kt

```diff
@@ -77,6 +78,30 @@ class ServiceErrorGenerator(
                 // operation errors
                 renderImplFrom(symbolProvider.symbolForOperationError(operationShape), operationShape.errors)
             }
+            // Every waiter error can be converted into service::Error
+            if (operations.any { it.hasTrait<WaitableTrait>() }) {
+                rustTemplate(
+                    """
+                    impl<O, E> #{From}<#{WaiterError}<O, E>> for Error
+                    where
+                        O: #{Debug} + #{Send} + #{Sync} + 'static,
+                        E: #{StdError} + #{Send} + #{Sync} + 'static,
+                    {
+                        fn from(err: #{WaiterError}<O, E>) -> Self {
+                            Error::Unhandled(#{Unhandled} {
```

Comment:
**question**: It looks like `Error::Unhandled` is marked `deprecated` in the generated code and is documented as "not forwards compatible". Is this what we want to map these to? (I don't have a better suggestion but I am curious on the rationale on why this is the right choice).

Reply:
That's to prevent customers from explicitly matching on it, not to prevent us from using it. If a customer matches `Unhandled` instead of `_`, then their code will break if a new error is modeled later.